src/pks/transport/transport_ats_pk.cc

```diff
@@ -466,7 +466,7 @@ void Transport_ATS::Initialize(const Teuchos::Ptr<State>& S)
 
   // source term initialization: so far only "concentration" is available.
   if (plist_->isSublist("source terms")) {
-    PK_DomainFunctionFactory<TransportDomainFunction> factory(mesh_);
+    PK_DomainFunctionFactory<TransportDomainFunction> factory(mesh_, S_);
```

Comment:
I just want to make sure this is the "right" State being provided (vs. S_next_).

What is done with this State?  Is the assumption this is used to grab time values at the "current" time not the "new" time?  I'm assuming it is, since transport is explicit, but just wanted to confirm.

Reply:
@lipnikov Can you confirm this?  I'm not sure what this state gets used for.

Would like to merge this ASAP since the code doesn't currently build.

Reply:
S_ is passed inside specific constructors, see set_state(S_) in each occasion. This prevent from running constructors and Init() with incorrect.undefined state.

Reply:
Ok, thanks